### ISSUE TITLE
我发现作者是通过在测试集上的结果选取最优模型，想问下，这么做合理吗？目前按照作者选择最优结果的方法，在resume数据集上只达到95.15%，没有达到95.5%。

### ISSUE THREAD

**Issue by wn1652400018** [LeeSureman/Flat-Lattice-Transformer#50]
1.最优结果的选择时候合理。
2.我总感觉作者transformer_encoder时的residual结构写的有问题，但是我把它修改后发现效果变差了。不知道为什么。
`class Layer_Process(nn.Module):
    def __init__(self, process_sequence, hidden_size, dropout=0,
                 use_pytorch_dropout=True):
        super().__init__()
        self.use_pytorch_dropout = use_pytorch_dropout
        self.process_sequence = process_sequence.lower()
        self.hidden_size = hidden_size
        self.dropout_rate = dropout
        if 'd' in self.process_sequence:
            if self.use_pytorch_dropout:
                self.dropout = nn.Dropout(dropout)
            else:
                self.dropout = MyDropout(dropout)
        if 'n' in self.process_sequence:
            self.layer_norm = nn.LayerNorm(hidden_size)

    def forward(self, inp): 
        output = inp
        for op in self.process_sequence: #process_sequence=’an‘
            if op == 'a':
                output = output + inp  #这里不是相当于将inp*2吗？
            elif op == 'd':
                output = self.dropout(output)
            elif op == 'n':
                output = self.layer_norm(output)

        return output`
3.修改了个别超参，如batch设置为5，k_proj修改为True，作者设置为false，另外在融合位置embed的时候ss，se，es，ee都使用了，作者的超参只使用了ss，ee，为了增大每个注意力头的大小稍微怎大了隐藏层大小。其中将k_proj修改为True就可以到达95.0%，使用4个相对位置融合感觉没有提升，增大隐藏层大小上升到95.15%。笔记本空间有限就没有继续增加隐藏层。

**Comment by wn1652400018** [comment]
@LeeSureman

**Comment by LeeSureman** [comment]
这个数据集比较小，可能需要多跑几遍。我是在dev集上选择最优模型的

**Comment by wn1652400018** [comment]
作者你好，我看见你在其他的issue里提到更正了residual结构后，效果有提升。可是我试了好多次效果都是下降的，不知道什么原因。不知道作者能不能跟新下这部分的代码。或许是我把这部分代码写错了。。。感谢大佬。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "李孝男"<notifications@github.com&gt;; 
发送时间: 2020年12月4日(星期五) 上午9:04
收件人: "LeeSureman/Flat-Lattice-Transformer"<Flat-Lattice-Transformer@noreply.github.com&gt;; 
抄送: "微冷/不觉寒"<1652400018@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [LeeSureman/Flat-Lattice-Transformer] 我发现作者是通过在测试集上的结果选取最优模型，想问下，这么做合理吗？目前按照作者选择最优结果的方法，在resume数据集上只达到95.15%，没有达到95.5%。 (#50)





 
这个数据集比较小，可能需要多跑几遍。我是在dev集上选择最优模型的

**Comment by wn1652400018** [comment]
对不起，我这里看错了。确实是按照dev来选择最优模型的。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "李孝男"<notifications@github.com&gt;; 
发送时间: 2020年12月4日(星期五) 上午9:04
收件人: "LeeSureman/Flat-Lattice-Transformer"<Flat-Lattice-Transformer@noreply.github.com&gt;; 
抄送: "微冷/不觉寒"<1652400018@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [LeeSureman/Flat-Lattice-Transformer] 我发现作者是通过在测试集上的结果选取最优模型，想问下，这么做合理吗？目前按照作者选择最优结果的方法，在resume数据集上只达到95.15%，没有达到95.5%。 (#50)





 
这个数据集比较小，可能需要多跑几遍。我是在dev集上选择最优模型的

**Comment by LeeSureman** [comment]
你邮箱多少啊，我私发你吧，我之前的代码每个数据集都固定了随机数种子来提升能复现论文报道结果的可能，如果直接在这上面更新，种子就没用了。最近没多余的卡来找另外的随机数种子了

**Comment by wn1652400018** [comment]
1652400018@qq.com。谢谢大佬。




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "李孝男"<notifications@github.com&gt;; 
发送时间: 2020年12月4日(星期五) 上午9:25
收件人: "LeeSureman/Flat-Lattice-Transformer"<Flat-Lattice-Transformer@noreply.github.com&gt;; 
抄送: "微冷/不觉寒"<1652400018@qq.com&gt;; "Author"<author@noreply.github.com&gt;; 
主题: Re: [LeeSureman/Flat-Lattice-Transformer] 关于residual结构的疑问。在resume数据集上只达到95.15%，还没有达到95.5%。 (#50)





 
你邮箱多少啊，我私发你吧，我之前的代码每个数据集都固定了随机数种子来提升能复现论文报道结果的可能，如果直接在这上面更新，种子就没用了。最近没多余的卡来找另外的随机数种子了

**Comment by nk2000** [comment]
> 我之前的代码每个数据集都固定了随机数种子来提升能复现论文报道结果的可能，如果直接在这上面更新，种子就没用了。最近没多余的卡来找另外的随机数种子了

nk1768@qq.com
也求一份随机种子，谢谢大佬。

**Comment by changle2018** [comment]
遇到同样的问题，折腾好久。同求一份，谢谢大佬。 邮箱：changle2003@126.com

**Comment by 18758014460** [comment]
可以也给我发一份吗，谢谢大佬。
3284784851@qq.com

**Comment by j-h015** [comment]
同求一份，谢谢大佬。 690096098@qq.com

**Comment by zhangliang-chn** [comment]
请问大佬后来复现了论文效果了，我现在在resume数据集上实验，不加BERT的效果只有94.8%左右，和论文的95.45%有差距，请问是什么原因呢？

**Comment by zhangliang-chn** [comment]
同求一份随机种子，谢谢大佬，zhangliangchn@foxmail.com
